core/controllers/classifier_test.py

```diff
@@ -119,3 +119,56 @@ def test_error_on_existing_classifier(self):
         classifier_services.create_classifier(self.job_id, self.classifier_data)
         self.post_json('/ml/trainedclassifierhandler', self.payload,
                        expect_errors=True, expected_status_int=500)
+
+
+class NextJobHandlerTest(test_utils.GenericTestBase):
```

Comment:
I think it would be a good idea to test all cases where Exceptions might be raised. Try changing the vm_id to default and set DEV_MODE to False. And similarly, test the other cases as well.